mono/mini/interp/transform.c

```diff
@@ -1386,12 +1406,25 @@ interp_handle_intrinsics (TransformData *td, MonoMethod *target_method, MonoClas
 	} else if (in_corlib && !strcmp (klass_name_space, "System") && !strcmp (klass_name, "RuntimeMethodHandle") && !strcmp (tm, "GetFunctionPointer") && csignature->param_count == 1) {
 		// We must intrinsify this method on interp so we don't return a pointer to native code entering interpreter
 		*op = MINT_LDFTN_DYNAMIC;
-	} else if (target_method->klass == mono_defaults.object_class) {
+	} else if (in_corlib && target_method->klass == mono_defaults.object_class) {
 		if (!strcmp (tm, "InternalGetHashCode"))
 			*op = MINT_INTRINS_GET_HASHCODE;
 #ifdef DISABLE_REMOTING
 		else if (!strcmp (tm, "GetType"))
 			*op = MINT_INTRINS_GET_TYPE;
+#endif
+#ifdef ENABLE_NETCORE
+		else if (!strcmp (tm, "GetRawData")) {
+			gint64 val = MONO_ABI_SIZEOF (MonoObject);
+			interp_add_ins (td, MINT_LDC_I8);
+			WRITE64_INS (td->last_ins, 0, &val);
```

Comment:
Will this work on 32 bit ?

Reply:
it would, with a warning though (stack type mismatch). let me fix it

Reply:
updated the PR.